adoc/admin-gpus.adoc

```diff
@@ -0,0 +1,163 @@
+[[gpus]]
+= NVIDIA GPUs
+
+SUSE's CaaS Platform supports scheduling GPU-dependent workloads on NVIDIA GPUs as a technical preview. This can be a powerful way to run compute-intensive workloads such as machine learning pipelines. This section illustrates how to prepare your host machine to expose GPU devices to your containers, and how to configure Kubernetes to schedule GPU-dependent workloads.
+
+== Prepare the host machine
+
+=== Install the GPU drivers
+
+Not every worker node in the cluster need have a GPU device present. On the nodes that do have one or more NVIDIA GPUs, install the drivers from NVIDIA's repository.
+
+[source,bash]
+----
+# zypper addrepo --refresh https://download.nvidia.com/suse/sle15sp2/ nvidia
+# zypper refresh
+# zypper install x11-video-nvidiaG05
+----
+
+[NOTE]
+====
+For most modern NVIDIA GPUs, the G05 driver will support your device. Some older devices may require the x11-video-nvidiaG04 driver package instead. Check NVIDIA's documentation for your GPU device model.
+====
+
+=== Install the OCI hooks
+
+OCI hooks are a way for vendors or projects to inject executable actions into the lifecycle of a container managed by the container runtime (runc). SUSE provides an OCI hook for NVIDIA that enable the container runtime and therefor the kubelet and the Kubernetes scheduler to query the host system for the presence of a GPU device and access it directly. Install the hook on the worker nodes with GPUs:
+
+[source,bash]
+----
+# zypper install nvidia-container-toolkit
+----
+
+=== Test a GPU container image
+
+At this point, you should be able to run a container image that requires a GPU and directly access the device from the running container, for example using podman:
+
+[source,bash]
+----
+# podman run docker.io/nvidia/cuda nvidia-smi
+----
+
+=== Troubleshooting
+
+At this point, you should be able to run a container image using a GPU. If that is not working, check the following:
+
+Ensure your GPU is visible from the host system:
+
+[source,bash]
+----
+# lspci | grep -i nvidia
+# nvidia-smi
+----
+
+Ensure the kernel modules are loaded:
+
+[source,bash]
+----
+# lsmod | grep nvidia
+----
+
+If they are not, try loading them explicitly and check dmesg for an error indicating why they are missing:
+
+[source,bash]
+----
+# nvidia-modprobe
+# dmesg | tail
+----
+
+== Configure Kubernetes
+
+=== Install the device plugin
+
+The Kubernetes device plugin framework allows the kubelet to advertise system hardware resources that the Kubernetes scheduler can then use as hints to schedule workloads that require such devices. The Kubernetes device plugin from NVIDIA allows the kubelet to advertise NVIDIA GPUs it finds present on the worker node. Install the device plugin using kubectl:
+
+[source,bash]
+----
+$ kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta6/nvidia-device-plugin.yml
```

Comment:
The source for the device plugin manifest is github, do we need to be concerned with requiring using a manifest not packaged by us?

Reply:
@qnetter is the above acceptable too you? this is the nvidia device plugin that lets you schedule gpu workloads. IMO it brings no value to package it ourselves.

Reply:
I see no problem with doing that. This is part of the official doc from NVIDIA.